gensim/models/ldamodel.py

```diff
@@ -498,7 +501,8 @@ def inference(self, chunk, collect_sstats=False):
             # The optimal phi_{dwk} is proportional to expElogthetad_k * expElogbetad_w.
             # phinorm is the normalizer.
             # TODO treat zeros explicitly, instead of adding 1e-100?
-            phinorm = np.dot(expElogthetad, expElogbetad) + 1e-100
+            eps = 1e-100 if self.dtype == np.float64 else (1e-35 if self.dtype == np.float32 else 1e-5)
```

Comment:
Easier to read and maintain as a mapping?

```python
dtype_to_eps = {
    np.float64: 1e-100,
    np.float32: 1e-35,
    np.float16: 1e-5,
}
```

And then also `if dtype not in dtype_to_eps: ...`